test/Streamly/Test/Prelude/Serial.hs

```diff
@@ -58,6 +60,38 @@ import Streamly.Internal.Data.Time.Clock (Clock(Monotonic), getTime)
 
 import Streamly.Test.Common
 import Streamly.Test.Prelude.Common
+import Data.Void (Void)
+import Control.Monad.IO.Class (MonadIO(liftIO))
+
+toList :: IS.Stream IO a -> IO [a]
+toList = S.fold FL.toList
+
+splitOn :: Monad m =>
+    (a -> Bool) -> FL.Fold m a b -> IS.Stream m a -> IS.Stream m b
+splitOn predicate f = IS.foldManyPost (FL.takeEndBy_ predicate f)
+
+splitOnSuffix :: Monad m =>
+    (a -> Bool) -> FL.Fold m a b -> IS.Stream m a -> IS.Stream m b
+splitOnSuffix predicate f = IS.foldMany (FL.takeEndBy_ predicate f)
+
+splitOnSuffixSeq' :: (MonadIO m, Unbox a1, Enum a1, Eq a1) =>
+    A.Array a1 -> FL.Fold m a1 a2 -> IS.Stream m a1 -> IS.Stream m a2
+splitOnSuffixSeq' patt f m =
+    IS.fromStreamD $ D.splitOnSuffixSeq False patt f (IS.toStreamD m)
```

Comment:
We should not use StreamD here, it is an internal module. Why not use the takeEndBySeq/takeEndBySeq_ folds?